### ISSUE TITLE
Parcel API: Allow specifying different sources per target

### ISSUE THREAD

**Issue by LekoArts** [parcel-bundler/parcel#8684]
# 🙋 feature request

The `package.json` API allows specifying different sources per target: https://parceljs.org/features/targets/#package.json%23targets.*.source

```json
{
  "targets": {
    "frontend": {
      "source": "app/index.html"
    },
    "backend": {
      "source": "api/index.js"
    }
  }
}
```

At the moment this isn't available with the [Parcel API](https://parceljs.org/features/parcel-api/).

## 🤔 Expected Behavior

With the Parcel API you're able to define the source per target in the same way that `entries` works.

```js
import {Parcel} from '@parcel/core';

let bundler = new Parcel({
  defaultConfig: '@parcel/config-default',
  mode: 'production',
  targets: {
    modern: {
      source: "**/*.js"
    },
    legacy: {
      source: "**/legacy/*.js"
    }
  }
});
```

## 😯 Current Behavior

Not available.

## 💁 Possible Solution

Basically the codepath in https://github.com/parcel-bundler/parcel/blob/f8d3fc30ca5b33d8f8674525f2a741d662c5986a/packages/core/core/src/requests/EntryRequest.js#L201-L237 also needs to happen for `this.options.targets` inside `resolveEntry` function.

Open question about this: What should happen with the top-level `entries`? Currently it's required but how would that play into this?

**Comment by blipper** [comment]
Updating the API to make it clear that this can't be done would save time!
